File: code-signal/the-core/area-1/8-phone-call.rb
def solution(min1, min2_10, min11, s)
  counter = 0
  while true
    case counter
    when 0 .. 1
      s -= min1
    when 2 .. 10
      s -= min2_10
    else
      s -= min11
    end
    if s < 0 and counter == 0
      return counter
    elsif s < 0
      return counter - 1
    elsif counter == 0
      counter = 2
    else
      counter += 1
    end
  end
end

File: code-signal/the-core/area-1/8-phone-call.js
function solution(min1, min2_10, min11, s) {
  if (s < min1) return 0;
  if (s === min1) return 1;

  let minutesCounter = 2;
  let tempS = s - min1
  while (true) {
    if (minutesCounter > 1 &&
      minutesCounter <= 10) {
      tempS -= min2_10
    } else {
      tempS -= min11
    }
    if (tempS < 0) {
      return minutesCounter - 1;
    }
    minutesCounter++
  }
}
